driver/ssh/config_test.go

```diff
@@ -22,15 +22,24 @@ func TestSSHConfigEncodeWithDSN(t *testing.T) {
 }
 
 func TestSSHConfigDecodeWithDSN(t *testing.T) {
-	encodedDSN := "postgres://ssh_user@ssh_host:22/user:password@host:5439/db?query1=val1&query2=val2&ssh_private_key=myprivate_key%26val"
-	config := &Config{}
-
-	whDSN, err := config.DecodeFromDSN(encodedDSN)
-	require.Nil(t, err)
-	require.Equal(t, "postgres://user:password@host:5439/db?query1=val1&query2=val2", whDSN)
-
-	require.Equal(t, config.Host, "ssh_host")
-	require.Equal(t, config.Port, 22)
-	require.Equal(t, config.User, "ssh_user")
-	require.Equal(t, string(config.PrivateKey), "myprivate_key&val")
+	t.Run("valid dsn", func(t *testing.T) {
+		encodedDSN := "postgres://ssh_user@ssh_host:22/user:password@host:5439/db?query1=val1&query2=val2&ssh_private_key=myprivate_key%26val"
+		config := &Config{}
+
+		whDSN, err := config.DecodeFromDSN(encodedDSN)
+		require.Nil(t, err)
+		require.Equal(t, "postgres://user:password@host:5439/db?query1=val1&query2=val2", whDSN)
+
+		require.Equal(t, config.Host, "ssh_host")
+		require.Equal(t, config.Port, 22)
+		require.Equal(t, config.User, "ssh_user")
+		require.Equal(t, string(config.PrivateKey), "myprivate_key&val")
+	})
+	t.Run("invalid dsn", func(t *testing.T) {
+		encodedDSN := "postgres://ssh_user@http://ssh_host/:22/user:password@host:5439/db?query1=val1&query2=val2&ssh_private_key=myprivate_key%26val"
+		config := &Config{}
+
+		_, err := config.DecodeFromDSN(encodedDSN)
+		require.Error(t, err)
```

Comment:
What error is expected here? We changed multiple ones. Are you sure this test is enough to cover all the failing scenarios?